### PR TITLE
[ML-Dataframe] change licensing check and plugin name/description

### PR DESCRIPTION
**Pull request by hendrikmuhs**
**Feature branch PR**

change the internal naming to data frame, adds its own license check and fixes tests

### CODE REVIEW

**Comment by hendrikmuhs** [comment]
investigating build errors, looks like I need further plumbing to get CI green.

File: x-pack/plugin/ml-feature-index-builder/src/main/java/org/elasticsearch/xpack/ml/featureindexbuilder/action/TransportPutFeatureIndexBuilderJobAction.java [hunk 1]
```diff
@@ -63,7 +63,7 @@ protected String executor() {
     protected void masterOperation(Request request, ClusterState clusterState, ActionListener<Response> listener) throws Exception {
 
         if (!licenseState.isMachineLearningAllowed()) {
-            listener.onFailure(LicenseUtils.newComplianceException(XPackField.FIB));
```
Comment (benwtrent) [inline x-pack/plugin/ml-feature-index-builder/src/main/java/org/elasticsearch/xpack/ml/featureindexbuilder/action/TransportPutFeatureIndexBuilderJobAction.java]: Why are we still checking `!licenseState.isMachineLearningAllowed()` in the line before this? We should be uniform in what license we require to use this feature, no?

Comment (hendrikmuhs) [reply x-pack/plugin/ml-feature-index-builder/src/main/java/org/elasticsearch/xpack/ml/featureindexbuilder/action/TransportPutFeatureIndexBuilderJobAction.java]: good catch! should be `isDataFrameAllowed()`

**Review by benwtrent — APPROVED** [review]

**Comment by MirAkram** [comment]
mj1265849@gmail.com
در تاریخ ۲۰ نوامبر ۲۰۱۸ ۷:۳۰ بعدازظهر، "Benjamin Trent" <
notifications@github.com> نوشت:

**Pull request merged**